Sources/libhostmgr/libhostmgr.swift

```diff
@@ -0,0 +1,312 @@
+    let remoteImages = try await remoteRepository.listImages(sortedBy: strategy)
+    let localImages = try localRepository.list()
+
+    return remoteImages
+        .filter(includingItemsIn: manifest)
+        .filter(excludingItemsIn: localImages.map(\.basename))
+}
+
+/// Calculates a list of local images that should be deleted because they're not part of the remote manifest
+///
+public func listLocalImagesToDelete(
+    localRepository: LocalVMRepository = LocalVMRepository(),
+    remoteRepository: RemoteVMRepository = RemoteVMRepository()
+) async throws -> [LocalVMImage] {
+    let manifest = try await remoteRepository.getManifest()
+    let localImages = try localRepository.list()
+
+    return localImages
+        .filter(excludingItemsIn: manifest)
+        .filter(excludingItemsIn: Configuration.shared.protectedImages)
+}
+
+// MARK: Virtual Machine Control
+public func lookupParallelsVMOrExit(
+    withIdentifier id: String,
+    parallelsRepository: ParallelsVMRepository = ParallelsVMRepository()
+) throws -> VM {
+    guard let parallelsVirtualMachine = try parallelsRepository.lookupVM(byIdentifier: id) else {
+        Console.crash(
+            message: "There is no VM with the name or UUID `\(id)` registered with Parallels",
+            reason: .parallelsVirtualMachineDoesNotExist
+        )
+    }
+
+    return parallelsVirtualMachine
+}
+
+public func startVM(name: String) async throws {
+    let startDate = Date()
+
+    guard let sourceVM = try LocalVMRepository().lookupVM(withName: name) else {
+        Console.crash(message: "VM \(name) could not be found", reason: .fileNotFound)
+    }
+
+    try resetVMStorage()
+
+    if sourceVM.state == .packaged {
+        try await unpackVM(name: name)
+        return try await startVM(name: name)
+    }
+
+    let destination = FileManager.default.temporaryFilePath(named: name + ".tmp.pvm")
+    try FileManager.default.removeItemIfExists(at: destination)
+    try FileManager.default.copyItem(at: sourceVM.path, to: destination)
+    Console.info("Created temporary VM at \(destination)")
+
+    guard let parallelsVM = try Parallels().importVM(at: destination)?.asStoppedVM() else {
+        Console.crash(message: "Unable to import VM: \(destination)", reason: .unableToImportVM)
+    }
+
+    Console.success("Successfully Imported \(parallelsVM.name) with UUID \(parallelsVM.uuid)")
+
+    try applyVMSettings([
+        .memorySize(Int(ProcessInfo().physicalMemory / 1024 / 1024) - 4096),  // We should make this configurable
+        .cpuCount(ProcessInfo().physicalProcessorCount),
+        .hypervisorType(.apple),
+        .networkType(.shared),
+        .isolateVM(.on),
+        .sharedCamera(.off)
+    ], to: parallelsVM)
+
+    try parallelsVM.start(wait: false)
+
+    let _: Void = try await withCheckedThrowingContinuation { continuation in
+        do {
+            try waitForVMStartup(parallelsVM)
+        } catch {
+            continuation.resume(with: .failure(error))
```

Comment:
Missing a return here? I think it's an error to call `resume(with:)` multiple times.

Reply:
Ah yep thanks!! Addressed in 6abc3dfdee8232e6ff9c6d2f03a75e4fb9bf6fab